server/core/include/irods/configuration_hook_manager.hpp

```diff
@@ -0,0 +1,65 @@
+#ifndef CONFIGURATION_HOOK_MANAGER_HPP
+#define CONFIGURATION_HOOK_MANAGER_HPP
+
+#include <vector>
+#include <string>
+#include <nlohmann/json.hpp>
+/// \file
+
+namespace irods::experimental::hooks{
+  /**
+
+   */
+  using hook =
+    std::function<void(const std::string&,const nlohmann::json&,const nlohmann::json&)>;
+  ///
+  ///Provides a mechanism to trigger the execution of code
+  ///when a value in the configuration changes.
+  ///
+  ///Hooks are responsible for determining 
+  class configuration_hook_manager{
+    configuration_hook_manager();
+    std::vector<hook> hooks_;
+  public:
+    static configuration_hook_manager& instance();
+    void add_hook(hook a);
+    void run_hooks(const std::vector<nlohmann::json> &changes);
+  };
+  /// CRTP base class to add some hook behavior.
+  /// 
+  /// Classes that derive from this should implement a function called
+  /// configuration_value_changed that takes a string and the old value and new value.
+  template<typename T> class hooked_resource{
```

Comment:
Where is this used?

Reply:
It isn't, I suppose that this should also probably be removed with the singleton class anyway :)